aiohttp/connector.py

```diff
@@ -392,11 +396,18 @@ def closed(self):
 
             try:
                 await fut
-            finally:
-                # remove a waiter even if it was cancelled
-                waiters.remove(fut)
-                if not waiters:
-                    del self._waiters[key]
+            except BaseException:
```

Comment:
Sorry, I don't understand why the deletion is moved from `finally` to `except`.
Why shouldn't we remove the waiter if **no** exceptions was raised?

Reply:
@asvetlov this is for two reasons: performance, and that if no exception is thrown the removal happened by the release method.